prov/sockets/src/sock_progress.c

```diff
@@ -2111,7 +2120,8 @@ int sock_pe_progress_rx_ctx(struct sock_pe *pe, struct sock_rx_ctx *rx_ctx)
 	struct dlist_entry *entry;
 	struct sock_pe_entry *pe_entry;
 
-	fastlock_acquire(&pe->lock);
+	if (fastlock_tryacquire(&pe->lock))
+		return 0;
```

Comment:
Another approach: We can add another lock for counter update/read operation.
So, we will never hit the hold-and-wait issue in manual-progress mode. 
Same for CQs.

Any thoughts?